modules/apps/27-interchain-accounts/keeper/account_test.go

```diff
@@ -67,3 +68,74 @@ func (suite *KeeperTestSuite) TestInitInterchainAccount() {
 		})
 	}
 }
+
+func (suite *KeeperTestSuite) TestInitChannel() {
+	var (
+		path   *ibctesting.Path
+		portID string
+	)
+
+	testCases := []struct {
+		name     string
+		malleate func()
+		expPass  bool
+	}{
+		{
+			"success", func() {}, true,
+		},
+		{
+			"port is not bound", func() {
+				portID = "invalid-portID"
+			}, false,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+
+		suite.Run(tc.name, func() {
+			suite.SetupTest() // reset
+			path = NewICAPath(suite.chainA, suite.chainB)
+			suite.coordinator.SetupConnections(path)
+			portID, _ = types.GeneratePortID(TestOwnerAddress, path.EndpointA.ConnectionID, path.EndpointA.Counterparty.ConnectionID)
+
+			// bind port and claim capability
+			cap := suite.chainA.GetSimApp().ICAKeeper.BindPort(suite.chainA.GetContext(), portID)
+			err := suite.chainA.GetSimApp().ICAKeeper.ClaimCapability(suite.chainA.GetContext(), cap, host.PortPath(portID))
+			suite.Require().NoError(err)
+
+			tc.malleate() // explicitly change fields in channel and testChannel
+
+			// get next channel seq
+			channelSequence := path.EndpointA.Chain.App.GetIBCKeeper().ChannelKeeper.GetNextChannelSequence(path.EndpointA.Chain.GetContext())
+
+			// Init new channel
+			err = suite.chainA.GetSimApp().ICAKeeper.InitChannel(suite.chainA.GetContext(), portID, path.EndpointA.ConnectionID)
+
+			if tc.expPass {
+				suite.Require().NoError(err)
+
+				// finish the channel handshake
+
+				// commit state changes for proof verification
+				path.EndpointA.Chain.App.Commit()
```

Comment:
maybe overkill finishing the handshake here?

Reply:
I have no strong preference, but might be better to split this off into its own test, ie one test doing table tests and another finishing the handshake in the happy case